test/shared/src/main/scala/zio/test/TestAspect.scala

```diff
@@ -303,9 +303,37 @@ object TestAspect extends TimeoutVariants {
       predicate: L => Boolean,
       spec: ZSpec[R, E, L, S]
     ): ZSpec[R, E, L, S] =
-      spec.transform[L, ZIO[R, TestFailure[E], TestSuccess[S]]] {
+      spec.transform[R, E, L, Either[TestFailure[Nothing], TestSuccess[S]]] {
         case c @ Spec.SuiteCase(_, _, _) => c
-        case Spec.TestCase(label, test)  => Spec.TestCase(label, if (predicate(label)) perTest(test) else test)
+        case Spec.TestCase(label, test) =>
+          Spec.TestCase(label, if (predicate(label)) from(perTest(to(test))) else test)
       }
+
+    /**
+     * These methods evidence an isomorphism between our previous encoding of
+     * `ZTest` and the new encoding. We can remove them once we refactor test
+     * aspects to use the new encoding
+     */
+    private def to[R, E, S](
```

Comment:
Will this be part of this pull request, or come along later?

Reply:
I was planning on doing it as part of this pull request, I just wanted to make sure we were aligned before I went through and changed all the test aspects. The one thing that may change this is if we are going to do a release today it could be nice to get this in so that projects that need the shared resource functionality can start using ZIO Test, even if we need to do a little cleanup on the backend afterwards.

Reply:
I will have this done shortly.